.github/actions/csv-vulnerability-filter/src/processing.ts

```diff
@@ -0,0 +1,341 @@
+    const uniqueId = Math.random().toString(36).substring(2, 8);
+    const filename = `${baseFilename}_${uniqueId}.csv`;
+    const outputPath = path.join(fallbackPath, filename);
+    core.info(`Writing output CSV to fallback location: ${outputPath}`);
+    try {
+      fs.writeFileSync(outputPath, csvString, 'utf8');
+      return outputPath;
+    } catch (error) {
+      core.error(`Failed to write output CSV file ${outputPath}: ${error instanceof Error ? error.message : String(error)}`);
+      throw new Error(`Failed to write output CSV to fallback location: ${outputPath}`);
+    }
+  }
+
+  const uniqueId = Math.random().toString(36).substring(2, 8);
+  const filename = `${baseFilename}_${uniqueId}.csv`;
+  const outputPath = path.join(tempDir, filename);
+
+  core.info(`Writing output CSV to: ${outputPath}`);
+  try {
+    fs.writeFileSync(outputPath, csvString, 'utf8');
+    return outputPath;
+  } catch (error) {
+    core.error(`Failed to write output CSV file ${outputPath}: ${error instanceof Error ? error.message : String(error)}`);
+    throw new Error(`Failed to write output CSV: ${outputPath}`);
+  }
+}
+
+export function writeOutputMarkdown(markdownString: string, baseFilename: string): string {
+  const tempDir = process.env.RUNNER_TEMP;
+  let outputPath: string;
+
+  if (!tempDir) {
+    core.warning("RUNNER_TEMP environment variable not defined. Using current directory for Markdown output.");
+    const fallbackPath = path.resolve('.');
+    const uniqueId = Math.random().toString(36).substring(2, 8);
+    const filename = `${baseFilename}_${uniqueId}.md`;
+    outputPath = path.join(fallbackPath, filename);
+    core.info(`Writing output Markdown to fallback location: ${outputPath}`);
+  } else {
+    const uniqueId = Math.random().toString(36).substring(2, 8);
+    const filename = `${baseFilename}_${uniqueId}.md`;
+    outputPath = path.join(tempDir, filename);
+    core.info(`Writing output Markdown to: ${outputPath}`);
+  }
+
+  try {
+    fs.writeFileSync(outputPath, markdownString, 'utf8');
+    return outputPath;
+  } catch (error) {
+    const errorMessage = error instanceof Error ? error.message : String(error);
+    core.error(`Failed to write output Markdown file ${outputPath}: ${errorMessage}`);
+    throw new Error(`Failed to write output Markdown to ${outputPath}: ${errorMessage}`);
+  }
+}
+
+// Helper to clean cell data for Markdown table
+function cleanCell(value: string | undefined | null, defaultValue = ''): string {
+  return value?.trim().replace(/\\/g, '\\\\').replace(/\|/g, '\\|').replace(/`/g, '\\`').replace(/\r?\n|\r/g, ' ') || defaultValue;
+}
+
+// Helper to format PURL (uses assumed column names)
+function formatPurl(pkgName: string | undefined | null, pkgVersion: string | undefined | null): string {
+  const cleanPkg = cleanCell(pkgName);
+  const cleanVer = cleanCell(pkgVersion);
+  if (!cleanPkg || !cleanVer) return '';
+  return `\`${cleanPkg}@${cleanVer}\``;
+}
```

Comment:
Unused? I think once you update this branch with main the linter will start running again and call out unused things like this

Reply:
Clean cell is used. Removed format purl